pkg/flow/node.go

```diff
@@ -5,56 +5,117 @@ import (
 
 	"github.com/jexia/semaphore/pkg/broker"
 	"github.com/jexia/semaphore/pkg/broker/logger"
+	"github.com/jexia/semaphore/pkg/functions"
 	"github.com/jexia/semaphore/pkg/references"
 	"github.com/jexia/semaphore/pkg/specs"
 	"github.com/jexia/semaphore/pkg/transport"
 	"go.uber.org/zap"
 )
 
+// NodeOptions represent a set of options that could be set through
+// option functions.
+type NodeOptions struct {
+	condition  *Condition
+	call       Call
+	functions  functions.Stack
+	rollback   Call
+	middleware *NodeMiddleware
+}
+
+// NodeOption is a wrapper function
+type NodeOption func(*NodeOptions)
+
+// WithCall sets the given call
+func WithCall(call Call) NodeOption {
+	return func(option *NodeOptions) {
+		option.call = call
+	}
+}
+
+// WithRollback sets the given call as rollback
+func WithRollback(call Call) NodeOption {
+	return func(option *NodeOptions) {
+		option.rollback = call
+	}
+}
+
+// WithCondition sets the given condition
+func WithCondition(condition *Condition) NodeOption {
+	return func(option *NodeOptions) {
+		option.condition = condition
+	}
+}
+
+// WithFunctions sets the given functions stack
+func WithFunctions(functions functions.Stack) NodeOption {
+	return func(option *NodeOptions) {
+		option.functions = functions
+	}
+}
+
+// WithNodeMiddleware sets the given middleware
+func WithNodeMiddleware(middleware *NodeMiddleware) NodeOption {
+	return func(option *NodeOptions) {
+		option.middleware = middleware
+	}
+}
+
+// NewNodeOptions constructs a new node options object and collects the options.
```

Comment:
This is the only one comment with dot (`.`) at the end
It is the only comment where linter would not complain :rofl: